integrations/getServices/index.test.ts

```diff
@@ -0,0 +1,75 @@
+import { getEnvValue } from "../../lib/env";
+import fetchApi from "../../lib/fetch";
+
+import { basicResponseDecoder } from "italia-ts-commons/lib/requests";
+import { PaginatedServiceTupleCollection } from "../../generated/definitions/backend/PaginatedServiceTupleCollection";
+
+const decoder = basicResponseDecoder(PaginatedServiceTupleCollection);
+
+describe("getServices", () => {
+  const host = getEnvValue("IO_BACKEND_HOST").getOrElseL(() => {
+    throw new Error(`required value dor "IO_BACKEND_HOST"`);
+  });
+  const basePath = getEnvValue("IO_BACKEND_BASEPATH").getOrElseL(() => {
+    throw new Error(`required value dor "IO_BACKEND_BASEPATH"`);
+  });
+  const sessionToken = getEnvValue("SPID_SESSION_TOKEN").getOrElseL(() => {
+    throw new Error(`required value dor "SPID_SESSION_TOKEN"`);
+  });
+  const endpoint = `${host}${basePath}/services`;
+
+  it("should return unauthorized in case it's called without sessionToken", async () => {
+    const headers = {};
+    const expectedHttpCode = 401;
+    const expectedBody = "Unauthorized";
+
+    const client = () => fetchApi(endpoint, { headers });
+    const response = await client();
+
+    expect(response.status).toBe(expectedHttpCode);
+
+    const body = await response.text();
+
+    expect(body).toEqual(expectedBody);
+  });
+
+  it("should return unauthorized in case it's called with an invalid sessionToken", async () => {
+    const headers = {};
+    const expectedHttpCode = 401;
+    const expectedBody = "Unauthorized";
+
+    const client = () => fetchApi(endpoint, { headers });
+    const response = await client();
+
+    expect(response.status).toBe(expectedHttpCode);
+
+    const body = await response.text();
+
+    expect(body).toEqual(expectedBody);
+  });
+
+  it("should correctly expose service list", async () => {
+    const headers = { Authorization: `Bearer ${sessionToken}` };
+    const expectedHttpCode = 200;
+
+    const client = (): Promise<Response> => fetchApi(endpoint, { headers });
+
+    const response = await client();
+    const decoded = await decoder(response);
+
+    expect(response.status).toBe(expectedHttpCode);
+
+    if (!decoded)
+      throw new Error(`Expected response to be in the correct format`);
+
+    decoded.fold(
+      e => {
+        console.error(e);
+        throw new Error(`Expected response to be in the correct format`);
```

Comment:
do not throw, just let the test fail

Reply:
Failing `expect`s actually throw. I don't know how to assert a code branch not to be executed (other than `expect(true).toBe(false)`).

Another solution would be implement `fold` callbacks as spies, and then check them to be called or not. But I think it's overkill other than very ugly.

Reply:
For the sake of code consistency, I may write a custom jest matcher that deals with a `fold`able structure. So we might have something like
```
expect(decoded).toBeRight()
```

Reply:
https://jest-bot.github.io/jest/docs/expect.html#expectassertionsnumber

Reply:
anyway, I find this kind of failure annoying: 
<img width="578" alt="Schermata 2020-03-05 alle 17 41 03" src="https://user-images.githubusercontent.com/265564/76003604-94db5280-5f08-11ea-9bdc-944006eeddcf.png">

We might find a method to express such failure in a more comunicative way. Something like "you passed this, but it doesn't decode into your type"

Reply:
I find it expressive (I read "expect the value to be a ServerInfo", what else ?)